WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java

```diff
@@ -1066,4 +1113,238 @@ private boolean isPluginStateChangedSinceLastConfigurationDispatch() {
+        if (mAutomatedTransferProgressDialog == null) {
+            mAutomatedTransferProgressDialog = new ProgressDialog(this);
+            mAutomatedTransferProgressDialog.setCancelable(false);
+            mAutomatedTransferProgressDialog.setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
+            mAutomatedTransferProgressDialog.setIndeterminate(false);
+            String message = getString(R.string.plugin_install_first_plugin_progress_dialog_title);
+            mAutomatedTransferProgressDialog.setMessage(message);
+        }
+        if (!mAutomatedTransferProgressDialog.isShowing()) {
+            mIsShowingAutomatedTransferProgress = true;
+            mAutomatedTransferProgressDialog.show();
+        }
+    }
+
+    /**
+     * Either Automated Transfer is completed or an error occurred.
+     */
+    private void cancelAutomatedTransferDialog() {
+        if (mAutomatedTransferProgressDialog != null && mAutomatedTransferProgressDialog.isShowing()) {
+            mAutomatedTransferProgressDialog.cancel();
+            mIsShowingAutomatedTransferProgress = false;
+        }
+    }
+
+    /**
+     * Automated Transfer successfully completed, the site has been refreshed and site plugins has been fetched. We can
+     * close the progress dialog, get the new version of the plugin from Store and refresh the views
+     */
+    private void automatedTransferCompleted() {
+        cancelAutomatedTransferDialog();
+        refreshPluginFromStore();
+        refreshViews();
+        showSuccessfulInstallSnackbar();
+        invalidateOptionsMenu();
+    }
+
+    /**
+     * Helper for if any of the FluxC Automated Transfer events fail. We are using a Toast for now, but the only likely
+     * error is the site missing a domain which will be implemented later on and will be handled differently.
+     */
+    private void handleAutomatedTransferFailed(String errorMessage) {
+        cancelAutomatedTransferDialog();
+        ToastUtils.showToast(this, errorMessage, Duration.LONG);
+    }
+
+    /**
+     * This is the first Automated Transfer FluxC event. It returns whether the site is eligible or not with a set of
+     * errors for why it's not eligible. We are handling a single error at a time right now, but the only likely error
+     * is the custom domain being missing from the business plan site which will be handled as a separate project soon.
+     *
+     * If the site is eligible, we'll initiate the Automated Transfer. Check out `onAutomatedTransferInitiated` for next
+     * step.
+     */
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onAutomatedTransferEligibilityChecked(OnAutomatedTransferEligibilityChecked event) {
+        if (isFinishing()) {
+            return;
+        }
+        if (!event.isEligible) {
+            String message =
+                    event.eligibilityErrors.isEmpty() ? getString(R.string.plugin_install_error_site_ineligible)
```

Comment:
I noticed while testing the related FluxC branch, `onAutomatedTransferEligibilityChecked()` showed an error for that site of mine that I added a domain to yesterday. I think we can ignore that error - it's a result of the domain process, I'm sure - but note that `event.eligibilityErrors` is null in this error situation, so the line `event.eligibilityErrors.isEmpty()` will result in a NPE.

I think it's worth checking for null here, and also fixing FluxC because `eligibilityErrors` is supposed to be `@NonNull`.

<img width="833" alt="screen shot 2018-04-10 at 9 25 45 am" src="https://user-images.githubusercontent.com/3903757/38578410-e0f4f0e6-3cd1-11e8-8422-9a387d2f13ea.png">

Reply:
> but note that `event.eligibilityErrors` is `null` in this error situation

That was an oversight and was fixed in the FluxC branch. It's now an empty list in case of an error. We still need to handle that case a bit better here though, which I'll try to do so :)